docs/src/explanations/Documentation-of-functions.md

```diff
@@ -0,0 +1,316 @@
+---
+
+layout: docs
+
+title:  "Documention of functions"
+
+section: "chisel3"
+
+---
+
+# Further Explanation for the := and  <> Operators
+
+Chisel contains two connection operators, `:=` and `<>`. This document serves for a deeper explanation of the differences of the two and when to use one or the other. The differences are demonstrated with experiments using Scastie examples + DecoupledIO examples.
+
+
+
+### Experiment Setup
+
+```scala mdoc
+// Imports used by the following examples
+import chisel3._
+import chisel3.util.DecoupledIO
+```
+
+The diagram for the experiment can be viewed **[here](https://docs.google.com/document/d/14C918Hdahk2xOGSJJBT-ZVqAx99_hg3JQIq-vaaifQU/edit?usp=sharing)**
+![Experiment Image](https://raw.githubusercontent.com/chipsalliance/chisel3/master/docs/src/images/chisel_01.png?sanitize=true)
+
+```scala mdoc:silent
+
+class Wrapper extends Module{
+  val io = IO(new Bundle {
+  val in = Flipped(DecoupledIO(UInt(8.W)))
+  val out = DecoupledIO(UInt(8.W))
+  })
+  val p = Module(new PipelineStage)
+  val c = Module(new PipelineStage) 
+  // connect Producer to IO
+  p.io.a <> io.in
+  // connect producer to consumer
+  c.io.a <> p.io.b
+  // connect consumer to IO
+  io.out <> c.io.b
+}
+class PipelineStage extends Module{
+  val io = IO(new Bundle{
+    val a = Flipped(DecoupledIO(UInt(8.W)))
+    val b = DecoupledIO(UInt(8.W))
+  })
+  io.b <> io.a
+}
+```
+Below we can see the resulting Verilog for this example:
+```scala modc
+ChiselStage.emitVerilog(new Wrapper)
+```
+## Concept 1: <> is Communicative
+
+
+
+This experiment is set up to test for the function of `<>` using the experiment above.
+
+Achieving this involves flipping the RHS and LHS of each arrow and seeing how '<>'  will react.
+( Scastie link for the experiment:https://scastie.scala-lang.org/Shorla/LVhlbkFQQnq7X3trHfgZZQ )
+
+
+
+
+```scala mdoc:silent:reset
+import chisel3._
+import chisel3.util.DecoupledIO
+
+class Wrapper extends Module{
+  val io = IO(new Bundle {
+  val in = Flipped(DecoupledIO(UInt(8.W)))
+  val out = DecoupledIO(UInt(8.W))
+  })
+  val p = Module(new PipelineStage)
+  val c = Module(new PipelineStage) 
+  // connect Producer to IO
+  io.in := p.io.a
+  // connect producer to consumer
+  p.io.b := c.io.a
+  // connect consumer to IO
+  c.io.b := io.out
+}
+
+class PipelineStage extends Module{
+  val io = IO(new Bundle{
+    val a = Flipped(DecoupledIO(UInt(8.W)))
+    val b = DecoupledIO(UInt(8.W))
+  })
+  io.b <> io.a
```

Comment:
This is the same module as above... To demonstrate commutativity, we want to swap the arguments:

```suggestion
  io.a <> io.b
```